framework/base/Module.php

```diff
@@ -626,14 +626,13 @@ public function createControllerByID($id)
             $className = substr($id, $pos + 1);
         }
 
-        if (!preg_match('%^[a-z][a-z0-9\\-_]*$%', $className)) {
-            return null;
-        }
-        if ($prefix !== '' && !preg_match('%^[a-z0-9_/]+$%i', $prefix)) {
+        if($this->incorrectClassNameOrPrefix($className, $prefix)) {
```

Comment:
`isIncorrectClassNameOrPrefix` is a better name. Also please fix formatting: `if<space>(`

Reply:
Done